internal/services/mariadb/mariadb_virtual_network_rule_resource.go

```diff
@@ -93,27 +92,12 @@ func resourceMariaDbVirtualNetworkRuleCreateUpdate(d *pluginsdk.ResourceData, me
 		},
 	}
 
-	if _, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ServerName, id.VirtualNetworkRuleName, parameters); err != nil {
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ServerName, id.VirtualNetworkRuleName, parameters)
+	if err != nil {
 		return fmt.Errorf("creating %s: %+v", id, err)
 	}
-
-	// Wait for the provisioning state to become ready
-	log.Printf("[DEBUG] Waiting for %s to become ready", id)
-	stateConf := &pluginsdk.StateChangeConf{
-		Pending:                   []string{"Initializing", "InProgress", "Unknown", "ResponseNotFound"},
-		Target:                    []string{"Ready"},
-		Refresh:                   mariaDbVirtualNetworkStateStatusCodeRefreshFunc(ctx, client, id.ResourceGroup, id.ServerName, id.VirtualNetworkRuleName),
-		MinTimeout:                1 * time.Minute,
-		ContinuousTargetOccurence: 5,
-	}
-	if d.IsNewResource() {
-		stateConf.Timeout = d.Timeout(pluginsdk.TimeoutCreate)
-	} else {
-		stateConf.Timeout = d.Timeout(pluginsdk.TimeoutUpdate)
-	}
```

Comment:
have you confirmed that this is no longer required? the LRO itself didn't correctly wait for the ProvisioningState at one point in time.

Reply:
Hi @tombuildsstuff , based on the tests, it seems that this is no longer required.

Reply:
@tombuildsstuff stateConf I added it again, please check again, thanks!